test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -185,12 +225,99 @@ var _ = ginkgo.Describe("ClusterQueue controller", func() {
 				return updatedCq.Status
 			}, framework.Timeout, framework.Interval).Should(gomega.BeComparableTo(kueue.ClusterQueueStatus{
 				UsedResources: emptyUsedResources,
-			}))
+				Conditions: []metav1.Condition{
+					{
+						Type:    kueue.ClusterQueueActive,
+						Status:  metav1.ConditionTrue,
+						Reason:  "Ready",
+						Message: "Can admit new workloads",
+					},
+				},
+			}, ignoreCQConditionTimestamps))
 			framework.ExpectPendingWorkloadsMetric(clusterQueue, 0, 0)
 			framework.ExpectAdmittedActiveWorkloadsMetric(clusterQueue, 0)
 		})
 	})
 
+	ginkgo.When("Reconciling clusterQueue status condition", func() {
+		var (
+			cq             *kueue.ClusterQueue
+			lq             *kueue.LocalQueue
+			wl             *kueue.Workload
+			cpuArchAFlavor *kueue.ResourceFlavor
+			cpuArchBFlavor *kueue.ResourceFlavor
+		)
+
+		ginkgo.BeforeEach(func() {
+			cq = testing.MakeClusterQueue("bar-cq").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor(flavorCPUArchA, "5").Max("10").Obj()).
+					Flavor(testing.MakeFlavor(flavorCPUArchB, "5").Max("10").Obj()).Obj()).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
+			lq = testing.MakeLocalQueue("bar-lq", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
+			wl = testing.MakeWorkload("bar-wl", ns.Name).Queue(lq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			framework.ExpectWorkloadToBeDeleted(ctx, k8sClient, wl, true)
+			framework.ExpectLocalQueueToBeDeleted(ctx, k8sClient, lq, true)
+			framework.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
+			framework.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, cpuArchAFlavor, true)
+			framework.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, cpuArchBFlavor, true)
+		})
+
+		ginkgo.It("Should update status conditions when flavors are created", func() {
```

Comment:
I would do this:
1. no flavors
2. create all flavors
3. remove one flavor

Reply:
Address this. It has the advantage that the condition flips from false to true and back to false.

Reply:
@alculquicondor, I tried to add test cases, as you said. Although, `3.remove one flavor` did not work fine for the following reasons.

1. Can not remove using flavor to block finalizers.
2. To avoid the first issue, I forcefully remove finalizers from the flavor and then successfully remove the flavor. Although, the clusterqueue-controller did not start reconciling and did not update the CQ status since the clusterqueue-controller does not watch ResourceFlavor in the below.

https://github.com/kubernetes-sigs/kueue/blob/ed57453794e53f6a74e7363be83c663873729f03/pkg/controller/core/clusterqueue_controller.go#L286-L291

I guess that clusterqueue-controller needs to watch ResourceFlavor resources. Do you have any other ideas to resolve this?

Reply:
> Uhm...... actually, this is working as intended. The resource flavor is in use, so it's not removed immediately thanks to the finalizer. We can leave this test as is.

Sounds good.

> However, how is the controller updating the status when the ResourceFlavors show up if the ClusterQueue controller is not watching ResourceFlavors? Let's understand this first. And then, we might need to add a custom watcher, similar to

@alculquicondor Yes, I agree with your idea. I try to dive into implementations to understand that. And then, If we need to add a custom watcher, add one.